service/history/historyEngine.go

```diff
@@ -1195,15 +1197,20 @@ func (e *historyEngineImpl) DescribeWorkflowExecution(
 		},
 	}
 
-	// TODO: we need to consider adding execution time to mutable state
-	// For now execution time will be calculated based on start time and cron schedule/retry policy
-	// each time DescribeWorkflowExecution is called.
-	startEvent, err := mutableState.GetStartEvent()
-	if err != nil {
-		return nil, err
+	// Old WorkflowExecutionInfo doesn't have execution time.
+	// For backwards compatibility continue to compute execution time.
+	// Remove this "if" block when it is ok to rely on executionInfo.ExecutionTime only (added 6/9/21).
+	if result.GetWorkflowExecutionInfo().GetExecutionTime() == nil {
+		// TODO: we need to consider adding execution time to mutable state
+		// For now execution time will be calculated based on start time and cron schedule/retry policy
+		// each time DescribeWorkflowExecution is called.
+		startEvent, err := mutableState.GetStartEvent()
+		if err != nil {
+			return nil, err
+		}
+		backoffDuration := timestamp.DurationValue(startEvent.GetWorkflowExecutionStartedEventAttributes().GetFirstWorkflowTaskBackoff())
+		result.WorkflowExecutionInfo.ExecutionTime = timestamp.TimePtr(timestamp.TimeValue(result.WorkflowExecutionInfo.GetStartTime()).Add(backoffDuration))
```

Comment:
could you move this logic within mutable state initialization logic here: https://github.com/temporalio/temporal/blob/v1.10.1/service/history/mutableStateBuilder.go#L266